File: 정건우/5주차/260126.java
//https://www.acmicpc.net/problem/23351
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class BOJ_S3_23351_물주기 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());
        int A = Integer.parseInt(st.nextToken());
        int B = Integer.parseInt(st.nextToken());

        PriorityQueue<Integer> pq = new PriorityQueue<>();

        N = N % A == 0 ? N/A : N/A + 1;

        for (int i = 0; i < N; i++) {
            pq.add(K);
        }

        int ans = 0;

        while (!pq.isEmpty() && ++ans <= pq.peek()) {
            pq.add(pq.poll()+B);
        }

        System.out.println(ans-1);

    }
}

File: 심수연/5주차/260126.py
# https://www.acmicpc.net/problem/11652

import sys
input = sys.stdin.readline

N = int(input())

dict = {}

for i in range(N):
  num = int(input())
  if num in dict:
    dict[num] += 1
  else:
    dict[num] = 1 # num 이 dict에 없다면 초기화

print(max(dict, key=lambda x: (dict[x], -x))) # 1순위: value값이 큰 것, 2순위: value값이 같다면 정수가 작은 것
